src/PageContent/RequestPickup/PickupSummary.js

```diff
@@ -2,26 +2,31 @@ import React from 'react';
 import './PickupSummary.css';
 import truck from '../../icons/green-truck.svg';
 import Map from '../../Map/Map.js';
-import { RequestRepeatType, RequestDurationType } from '../../Enums.js';
+import { RequestRepeatType, RequestEndCriteriaType, StringFormat } from '../../Enums.js';
+import moment from 'moment';
 
 class PickupSummary extends React.Component {
 
     render() {
+        let dayOfWeek = moment(this.props.request.startTimestamp).format(StringFormat.WEEKDAY);
         var repeatMap = {
-            [RequestRepeatType.WEEKLY]: 'every ' + this.props.dayOfWeek,
-            [RequestRepeatType.BIWEEKLY]: 'every other ' + this.props.dayOfWeek,
+            [RequestRepeatType.WEEKLY]: 'every ' + dayOfWeek,
+            [RequestRepeatType.BIWEEKLY]: 'every other ' + dayOfWeek,
             // TODO: calculate nth day of month
             [RequestRepeatType.MONTHLY]: 'monthly',
         };
 
         var durationText = '';
-        if (this.props.request.duration.type === RequestDurationType.RECUR) {
-            durationText = this.props.request.duration.value + ' pickups requested';
+        if (this.props.request.endCriteria.type === RequestEndCriteriaType.OCCUR) {
+            durationText = (parseInt(this.props.request.endCriteria.value, 10)) + ' pickups requested';
```

Comment:
Since our only use case here is displaying the number, and js is weakly typed, we could simply display it without parsing it into an int.

Reply:
ah you're right. This was a remnant from when it was still number of recurrences, not occurrences, so I had to parse the int so I could add `+1` to it to get the total number of repeats. But now since it's number of occurrences you're right, we can just print the string.